File: app/flutter/lib/domain/chat/vision_type_enum.dart


enum VisionTypeEnum {
  unknown("null"),
  image("image"),
  imageList("image_list"),
  video("video");

  final String code;
  const VisionTypeEnum(this.code);

  static VisionTypeEnum getVisionType(String code) {
    return values.firstWhere(
          (type) => type.code == code,
      orElse: () => VisionTypeEnum.unknown,
    );
  }

  // 为了方便使用，添加一个静态的默认值
  static const VisionTypeEnum defaultValue = VisionTypeEnum.image;

  // 扩展方法：判断是否为有效类型
  bool get isValid => this != VisionTypeEnum.unknown;

  // 转为字符串（用于序列化）
  @override
  String toString() => code;
}

File: project/app/android/app/src/main/java/com/magicvector/domain/constant/chat/VisionTypeEnum.kt
package com.magicvector.domain.constant.chat

// image, image list, video
enum class VisionTypeEnum(val code: String) {
    UNKNOWN("null"),
    IMAGE("image"),
    IMAGE_LIST("image_list"),
    VIDEO("video"),
    ;

    companion object {
        fun getVisionType(code: String): VisionTypeEnum {
            return entries.find { it.code == code } ?: UNKNOWN
        }
    }
}